### DISCUSSION TITLE
Questions about flashrom plugin

### DISCUSSION THREAD

**Discussion by tcyrus** [fwupd/fwupd#5492] [General]
I'm trying to make a firmware archive for my hardware ([GA-G97N](https://github.com/tcyrus/custom-lvfs/tree/main/com.gigabyte.GA-G97N.ami-flashrom)).

I know that Gigabyte is on the list of vendors, but it doesn't seem like there's an easy way to reach out to them, so I wanted to take a crack at it first.

I found certain parts of the documentation to be confusing. Some of these are specific to the `flashrom` plugin, but I'm not sure.

It's entirely possible that all of these questions are answered in the documentation and I managed to gloss over it.

- Are the `flashrom` GUIDs the same as the HardwareIDs?
- Is it normal for `sudo fwupdtool --plugin-whitelist flashrom get-devices --verbose` to return `No detected devices`?
- Is `<firmware type="flashed">` supposed to be the GUID/HWID present after flashing the hardware?
  - If so, why isn't `HardwareID-2` in the [GUID list](https://github.com/fwupd/fwupd/tree/main/plugins/flashrom#guid-generation) for `flashrom`?
- How do you use `<requires>` to make sure you're not flashing older firmware?

```shell
$ fwupdmgr --version
runtime   org.freedesktop.fwupd         1.7.9
runtime   com.dell.libsmbios            2.4
compile   org.freedesktop.gusb          0.3.4
runtime   org.kernel                    5.15.0-56-generic
compile   com.hughsie.libjcat           0.1.4
compile   org.freedesktop.fwupd         1.7.9
runtime   org.freedesktop.gusb          0.3.4
```

**Comment by hughsie** [comment]
> but it doesn't seem like there's an easy way to reach out to them

If it helps, I have the same problem too.

> Some of these are specific to the flashrom plugin

Any reason you want to use flashrom and not the (mostly standard) UEFI capsule plugin?

> Are the flashrom GUIDs the same as the HardwareIDs?

Does https://fwupd.github.io/libfwupdplugin/flashrom-README.html help?

> How do you use <requires> to make sure you're not flashing older firmware?

Does https://lvfs.readthedocs.io/en/latest/metainfo.html#adding-restrictions help?

**Reply by tcyrus** [reply]
> If it helps, I have the same problem too.

Well that's reassuring 🤣

> Any reason you want to use flashrom and not the (mostly standard) UEFI capsule plugin?

For this specific board, I believe that the flashrom plugin is the right one.

The payload provided by the manufacturer (Gigabyte) uses `EfiFlash.exe`, which seems to be similar to how `flashrom` works. Also, the payload matches the format of other flashrom payloads (the specific one I compared against was `com.starlabs.B6-I.ami-flashrom`) when viewed using something like UEFITool.

Also, I get the `UEFI capsule updates not available or enabled in firmware setup` message on my hardware.

As for the details related to the flashrom plugin and the Metainfo, I tried reading the docs before asking, but it's possible that I misunderstood something.

**Reply by hughsie** [reply]
> uses EfiFlash.exe, which seems to be similar to how flashrom works

Yes and no. In EFI mode the SPI is probably *unlocked* but in userspace it's probably *locked*. What does `fwupdmgr security` say about BIOSWE and BLE?

> UEFI capsule updates not available or enabled in firmware setup

Can you enable them in the setup menu of the BIOS? Sometimes they're called weird name like "Windows Update Firmware ESRT" and stuff like that.

**Reply by tcyrus** [reply]
Needed to force the check, but BIOSWE (SPI write) is Disabled and BLE (SPI lock) is Disabled.
I believe I checked the BIOS settings the first time I got the UEFI capsule message, but I'll check again.
